src/XR/features/WebXRLayers.ts

```diff
@@ -242,6 +257,15 @@ export class WebXRLayers extends WebXRAbstractFeature {
         return new WebXRProjectionLayerWrapper(projLayer, this._xrWebGLBinding);
     }
 
+    /**
+     * Add a new layer to the already-existing list of layers
+     * @param wrappedLayer the new layer to add to the existing ones
+     */
+    public addXRSessionLayer(wrappedLayer: WebXRLayerWrapper) {
+        this._existingLayers.push(wrappedLayer);
+        this.setXRSessionLayers(this._existingLayers);
```

Comment:
Should `setXRSessionLayers` set `this._existingLayers`? 
This doesn't seem like a function that will be called very often so it feels like you could do something like `this.setXRSessionLayers([...this._existingLayers, wrappedLayer])` and then in `setXRSessionLayers`, `this._existingLayers = wrappedLayers.slice(0)`

Reply:
I thought about it. At the beginning I exposed the existing layers, and then I thought the flow is - create a wrapper (and have a reference to it) and then add it to the session. I already have the wrapper (and am using it somewhere else), I don't know when I will need the entire array. Apart from when I want to remove the "base" layer, but that would stop rendering. What do you think?

Reply:
(BTW - I resolved the other conversation. I keep this one open for further discussion on this subject)